### DISCUSSION TITLE
Change title color, when popup is opened

### DISCUSSION THREAD

**Discussion by fury21** [LeoNatan/LNPopupController#568] [General] [answered]
Hello. How to change the title, subtitle and background color of the mini-player in ios when it is open? Now I adjust the colors before opening the mini-player and everything works for me, but changing the color does not work if the mini-player is already open.

private func setupPlayerPopUpBar() {
        let appearance = LNPopupBarAppearance()
        let titleTextAttributesForegroundColor = UIColor.Scheme.Label.text8()
        let subtitleTextAttributesForegroundColor = UIColor.Scheme.Label.text9()
        
        
        appearance.__titleTextAttributes = [.font: UIFont(name: "Jost-SemiBold", size: 14)!,
                                            .foregroundColor: titleTextAttributesForegroundColor]
        
        appearance.__subtitleTextAttributes = [.font: UIFont(name: "Jost-Regular", size: 12)!,
                                               .foregroundColor: subtitleTextAttributesForegroundColor]
        
        //        appearance.backgroundEffect = UIBlurEffect(style: .dark)
        appearance.backgroundColor = .Scheme.View.background10()
        
        let appearanceProxy = LNPopupBar.appearance(whenContainedInInstancesOf: [UINavigationController.self])
        appearanceProxy.inheritsAppearanceFromDockingView = false
        appearanceProxy.standardAppearance = appearance
        appearanceProxy.tintColor = .pink1_FF0B77()
        appearanceProxy.progressViewStyle = .top
        
        popupBar.barStyle = .compact
        
        popupInteractionStyle = .drag
        popupBar.imageView.contentMode = .scaleAspectFill
    }

**Comment by LeoNatan** [comment, marked as answer]
You should not be using the old appearance proxy API. Instead, set the properties to the bar’s appearance directly or to a popup item.

**Reply by fury21** [reply]
Give me an example please. I have ios project 14

**Reply by LeoNatan** [reply]
I suggest you look in the example app for how to use the API.

**Comment by fury21** [comment]
> I suggest you look in the example app for how to use the API.

I really don’t understand how to do this and I’ve already wasted a lot of time, please help me

**Reply by LeoNatan** [reply]
You are already presenting the popup bar on some controller. So access the popup bar property of that controller at runtime and modify the appearance, either before presenting the bar, or after.

**Reply by LeoNatan** [reply]
There is an example of code here:

https://github.com/LeoNatan/LNPopupController?tab=readme-ov-file#popup-bar-customization
